### PR TITLE
Generate usage documentation automatically

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [ ] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the section below.

### Changes description
...

### CODE REVIEW

File: src/snowcli/cli/app.py [hunk 1]
```diff
@@ -48,6 +57,14 @@ def default(
         callback=_version_callback,
         is_eager=True,
     ),
+    docs: bool = typer.Option(
+        None,
+        "--docs",
+        hidden=True,
+        help="Prints version of the snowcli",
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/app.py:64]: `Generate documentation` ?

Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/app.py]: Good catch 👌

**Review by sfc-gh-astus — DISMISSED** [review]

**Review by sfc-gh-astus — DISMISSED** [review]

File: tests/test_main.py [hunk 2]
```diff
@@ -49,3 +55,32 @@ def test_info_callback(runner):
         {"key": "version", "value": VERSION},
         {"key": "default_config_file_path", "value": str(cli_config.file_path)},
     ]
+
+
+def test_all_commands_has_proper_documentation():
```
Comment (sfc-gh-pjob) [inline tests/test_main.py:60]: Very good idea about this test!

File: src/snowcli/docs/generator.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from jinja2 import FileSystemLoader, Environment
+from typer.core import TyperArgument
+
+log = logging.getLogger(__name__)
+
+CMD_USAGE_TMPL = "usage.rst.jinja2"
+OVERVIEW_TMPL = "overview.rst.jinja2"
+
+
+def generate_docs(root: Path, command: Command, cmd_parts: Optional[List] = None):
```
Comment (sfc-gh-pjob) [inline src/snowcli/docs/generator.py:17]: Nice! Great job.

File: src/snowcli/docs/templates/usage.rst.jinja2 [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{{ help }}
+
+Syntax
+===============================================================================
+
+.. code-block:: console
```
Comment (sfc-gh-pjob) [inline src/snowcli/docs/templates/usage.rst.jinja2:6]: I wonder if we can describe somewhere how we can preview generated docs. Intellij's preview doesn't look good:

Screenshot 2023-08-01 at 00 04 02

**Review by sfc-gh-pjob — APPROVED** [review]

File: src/snowcli/docs/templates/usage.rst.jinja2 (hunk 4 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/docs/templates/usage.rst.jinja2:6]: Personally I use raw rst. I think proper preview will be part of next steps when we connect with the rest of the docs

**Pull request merged**